articles/200_migration_guide_from_ros1.md

```diff
@@ -139,12 +213,91 @@ Shared pointer types are provided as typedefs within the message structs: `my_in
 
 For more details please see the article about the [generated C++ interfaces](http://design.ros2.org/articles/generated_interfaces_cpp.html).
 
+The migration requires includes to change by:
+
+- insert the subfolder `msg` between the package name and message datatype
+- Change the included filename from CamelCase to underscore separation
+- Change from `*.h` to `*.hpp`
+
+``` cpp
+// ROS 1 style is in comments, ROS 2 follows, uncommented.
+// # include <geometry_msgs/PointStamped.h>
+#include <geometry_msgs/msg/point_stamped.hpp>
+
+// geometry_msgs::PointStamped point_stamped;
+geometry_msgs::msg::PointStamped point_stamped;
+```
+
+The migration requires code to insert the `msg` namespace into all instances.
+
+#### Usages of ros::Time
+
+<div class="alert alert-warning" markdown="1">
+ <b>TODO(tfoote) There is no direct replacement for ros::Time yet we expect to have one in the future. </b>
+</div>
+Under the hood we expect to leverage the cross platform `std::chrono` library.
+
+Currently for usages of `ros::Time`:
+
+- replace all instances of `ros::Time` with `builtin_interfaces::msg::Time`
+- Convert all instances of `nsec` to `nanosec`
+- Convert all single argument double constructors to bare constructor + assignment
+
+Field values do not get initialized to zero when constructed.
+You must make sure to set all values instead of relying on them to be zero.
```

Comment:
This is true for plain messages. However `ros::Time` was special and did zero it on construction: https://github.com/ros/roscpp_core/blob/indigo-devel/rostime/include/ros/time.h#L133